### PR TITLE
Dont write twice

### PR DESCRIPTION
**Pull request by Chronojam**
Added sorting for comparison

### CODE REVIEW

**Comment by swagatha-christie** [comment]
By analyzing the blame information on this pull request, we identified @bluepeppers to be a potential reviewer

File: main.go [hunk 1]
```diff
@@ -32,10 +34,25 @@ type SearchConfig struct {
 }
 
 type DiscoveryTarget struct {
-	Targets []string          `yaml:"targets"`
+	Targets []string           `yaml:"targets"`
```
Comment (bluepeppers) [inline main.go]: gofmt

File: main.go [hunk 2]
```diff
@@ -178,26 +195,37 @@ func findInstanceIP(instance *compute.Instance) (string, error) {
...
+	t := DiscoveryTargets(targets)
+	sort.Sort(t)
+
+	newData, err := yaml.Marshal(t)
 	if err != nil {
-		return errors.Wrap(err, "Failed to open output file")
+		return errors.Wrap(err, "Failed to Marshal YAML")
+	}
+
+	currentData, _ := ioutil.ReadFile(targetFile)
```
Comment (bluepeppers) [inline main.go]: What happens if the file does not exist here? Could we test that?

File: main.go [hunk 3]
```diff
@@ -32,10 +34,25 @@ type SearchConfig struct {
...
+	return len(dt)
+}
+
+func (dt DiscoveryTargets) Less(i, j int) bool {
+	return dt[i].Targets[0] < dt[j].Targets[0]
+}
+
+func (dt DiscoveryTargets) Swap(i, j int) {
+	dt[i], dt[j] = dt[j], dt[i]
+}
```
Comment (bluepeppers) [inline main.go]: Compact this a bit

**Pull request closed without merging**